src/main/java/de/tum/in/www1/artemis/domain/hestia/TestwiseCoverageReportEntry.java

```diff
@@ -0,0 +1,70 @@
+package de.tum.in.www1.artemis.domain.hestia;
+
+import javax.persistence.*;
+
+import org.hibernate.annotations.Cache;
+import org.hibernate.annotations.CacheConcurrencyStrategy;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import de.tum.in.www1.artemis.domain.DomainObject;
+import de.tum.in.www1.artemis.domain.ProgrammingExerciseTestCase;
+
+/**
+ * A single entry from testwise coverage report by file path and consecutive executed code block.
+ * A block is represented by the start line and the length (i.e. number of lines) of the block.
+ */
+@Entity
+@Table(name = "testwise_coverage_report_entry")
+@Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public class TestwiseCoverageReportEntry extends DomainObject {
```

Comment:
This is the only time you actually use Teswise in the name of one of the domain objects.
Not really a big issue, but definitely an inconsistency.

Reply:
The idea behind this idea is to design it in a way that support for other coverage tools could be added easily.

All coverage analyses have some aggregated statistics concerning the whole build (CoverageReport) and some statistics concerning single files (CoverageFileReport). But not all analyses map lines to test cases, therefore the distinction is made only here.